src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/CertificateRegistry.java

```diff
@@ -5,155 +5,81 @@
-import com.aws.greengrass.util.Digest;
-import com.aws.greengrass.util.Utils;
+import org.bouncycastle.util.encoders.Hex;
+import software.amazon.awssdk.utils.StringInputStream;
 
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.time.Instant;
 import java.util.Collections;
 import java.util.LinkedHashMap;
 import java.util.Map;
-import java.util.Optional;
-import javax.inject.Inject;
 
 
 public class CertificateRegistry {
     private static final Logger logger = LogManager.getLogger(CertificateRegistry.class);
-    // holds mapping of certificateHash (SHA-256 hash of certificatePem) to IoT Certificate Id;
+    // holds mapping of certificate ID -> last valid instant
     // size-bound by default cache size, evicts oldest written entry if the max size is reached
-    private final Map<String, String> registry = Collections.synchronizedMap(
-            new LinkedHashMap<String, String>(RegistryConfig.REGISTRY_CACHE_SIZE, 0.75f, false) {
+    private final Map<String, Instant> registry = Collections.synchronizedMap(
+            new LinkedHashMap<String, Instant>(RegistryConfig.REGISTRY_CACHE_SIZE, 0.75f, false) {
                 @Override
                 protected boolean removeEldestEntry(Map.Entry eldest) {
                     return size() > RegistryConfig.REGISTRY_CACHE_SIZE;
                 }
             });
 
-    private final IotAuthClient iotAuthClient;
-
     /**
-     * Constructor.
-     *
-     * @param iotAuthClient IoT Auth Client
+     * Retrieve certificate by certificate pem.
+     * @param certificatePem cert pem
+     * @return certificate object
      */
-    @Inject
-    public CertificateRegistry(IotAuthClient iotAuthClient) {
-        this.iotAuthClient = iotAuthClient;
-    }
-
-    /**
-     * Returns whether the provided certificate is valid and active.
-     * Returns valid locally registered result when IoT Core cannot be reached.
-     *
-     * @param certificatePem Certificate PEM
-     * @return true if the certificate is valid and active.
-     * @throws CloudServiceInteractionException if the certificate cannot be validated
-     */
-    public boolean isCertificateValid(String certificatePem) {
-        try {
-            Optional<String> certId = fetchActiveCertificateId(certificatePem);
-            updateRegistryForCertificate(certificatePem, certId);
-            return certId.isPresent();
-        } catch (CloudServiceInteractionException e) {
-            return getAssociatedCertificateId(certificatePem)
-                    .map(certId -> true)
-                    .orElseThrow(() -> e);
-        }
+    public Certificate getCertificateByPem(String certificatePem) {
+        return getCertificateById(getCertificateId(certificatePem));
```

Comment:
Possible NPE here. `getCertificateId()` returns `null` in case of `NoSuchAlgorithmException | CertificateException | UnsupportedEncodingException` while calculating certificate id.